web_ui/src/hooks/use-load-ai-webworker/use-load-ai-webworker.hook.ts

```diff
@@ -15,7 +15,7 @@ export const useLoadAIWebworker = <T extends AlgorithmType>(algorithmType: T) =>
             const baseWorker = getWorker(algorithmType);
             const worker = wrap<MapAlgorithmToWorker[T]>(baseWorker);
 
-            await worker.waitForOpenCV();
+            await worker.loadOpenCV();
```

Comment:
Something for later: I think we can change this hook so that it only calls the algorithm's factory method, e.g. ,
```
const worker = await(createInferenceImage());
```

and have that factory method be responsible for waiting for open cv, instead of having this hook do this.

My motivation for this is that this could then allow some workers to skip loading opencv, for instance if they don't need it.

Reply:
Yeah agree, when developing i went back and forth with this tbh. Ended up merging opencvloader and waitforopencv to delegate everything to the worker and in one single method to make it easier. The good news is that i can make the change you mentioned pretty easily now

Reply:
Updated